ete3/coretype/tree.py

```diff
@@ -146,9 +146,12 @@ def _set_up(self, value):
     def _get_children(self):
         return self._children
     def _set_children(self, value):
-        if type(value) == list and \
-           len(set([type(n)==type(self) for n in value]))<2:
-            self._children = value
+        if type(value) == list:
+            for n in value:
+                if type(n) == type(self):
+                    self._children = value
```

Comment:
This is incorrect, you are setting `value` in every iteration.  It should be something like:
``` 
for n in value:
  if type(n) != type(self):
    raise TreeError("Incorrect child node type")
  self._children = value
```